src/utilities/util_printlog.c

```diff
@@ -24,21 +24,47 @@ util_printlog(WT_SESSION *session, int argc, char *argv[])
     WT_DECL_RET;
     uint32_t flags;
     int ch;
+    // int start_lsn = 0;
+    WT_LSN start_lsn;
+    WT_LSN end_lsn;
+    uint32_t start_lsnfile;
+    uint32_t start_lsnoffset;
+    uint32_t end_lsnfile;
+    uint32_t end_lsnoffset;
     char *ofile;
+    char *start_str;
+    int n_args;
+    bool start_set = false;
+    bool end_set = false;
 
     flags = 0;
     ofile = NULL;
-    while ((ch = __wt_getopt(progname, argc, argv, "f:mx")) != EOF)
+    while ((ch = __wt_getopt(progname, argc, argv, "f:l:mx")) != EOF)
         switch (ch) {
         case 'f': /* output file */
             ofile = __wt_optarg;
+            printf("Outfile: %s\n", ofile);
             break;
         case 'm': /* messages only */
             LF_SET(WT_TXN_PRINTLOG_MSG);
             break;
         case 'x': /* hex output */
             LF_SET(WT_TXN_PRINTLOG_HEX);
             break;
+        case 'l':
+            start_str = __wt_optarg;
+            n_args = sscanf(start_str, "%" SCNu32 ",%" SCNu32 ",%" SCNu32 ",%" SCNu32,
+              &start_lsnfile, &start_lsnoffset, &end_lsnfile, &end_lsnoffset);
+            if (n_args == 2) {
+                WT_SET_LSN(&start_lsn, start_lsnfile, start_lsnoffset);
+                start_set = true;
+            } else if (n_args == 4) {
+                WT_SET_LSN(&start_lsn, start_lsnfile, start_lsnoffset);
+                WT_SET_LSN(&end_lsn, end_lsnfile, end_lsnoffset);
+                start_set = true;
+                end_set = true;
+            }
```

Comment:
Should have an error case here, if n_args isn't 2 or 4.

Reply:
Is there a particular way I should handle errors?

Reply:
In that situation you can just return (usage()).

Reply:
Would you be able to elaborate on that?